test/routes/origins.js

```diff
@@ -1,10 +1,56 @@
 // Copyright (c) The Linux Foundation and others. Licensed under the MIT license.
 // SPDX-License-Identifier: MIT
 
-const { expect } = require('chai')
+const { expect, assert } = require('chai')
 const httpMocks = require('node-mocks-http')
 const sinon = require('sinon')
 const originCondaRoutes = require('../../routes/originConda')
+const proxyquire = require('proxyquire')
+const fs = require('fs')
+
+describe('Pypi origin routes', () => {
+  let router
+  let requestPromiseStub
+  const fixturePath = 'test/fixtures/origins/pypi'
+  beforeEach(() => {
+    requestPromiseStub = options => {
```

Comment:
Can we set up the requestPromiseStub for each test case separate, so that it is easier to ready?  Let me know what you think. 
```
describe('Pypi origin routes', () => {
  let router
  let requestPromiseStub
  const fixturePath = 'test/fixtures/origins/pypi'
  beforeEach(() => {
    requestPromiseStub = sinon.stub()
    const createRoute = proxyquire('../../routes/originPyPi', { 'request-promise-native': requestPromiseStub })
    router = createRoute(true)
  })

  afterEach(function () {
    sinon.restore()
  })

  it('should return a valid response when a valid package is provided as input', async () => {
    requestPromiseStub.returns({ body: loadFixture(`${fixturePath}/pandas.json`), statusCode: 200 })
    const response = await router._getPypiData('pandas')
    expect(response.body.info.name).to.be.equal('pandas')
  })
  it('should return an empty response when a missing package is provided as input', async () => {
    requestPromiseStub.throws({ body: { message: 'Not Found' }, statusCode: 404 })
    expect(await router._getPypiData('pand')).to.be.deep.equal([])
  })
  it('should return a valid error message when an error other than 404 occurs', async () => {
    requestPromiseStub.throws({ statusCode: 400 })
    try {
      await router._getPypiData('pan')
      //Fail the test case if the above statement does not throw error
      assert.fail('should have thrown error')
    } catch (error) {
      expect(error.statusCode).to.be.equal(400)
    }
  })
})
```

Reply:
Also see See
[Refactor test · qtomlinson/service@fbf3736 (github.com)](https://github.com/qtomlinson/service/commit/fbf3736fa6f6e7b9392fe109751d8ee80c95f299)
for pypi fix.

Reply:
As per our discussion, the location of assert.fail needs to be adjusted.